arangod/Aql/AqlValue.cpp

```diff
@@ -1473,11 +1764,26 @@ bool std::equal_to<arangodb::aql::AqlValue>::operator()(arangodb::aql::AqlValue
   if (type != b.type()) {
     return false;
   }
-  if (type == arangodb::aql::AqlValue::VPACK_INLINE) {
-    return arangodb::velocypack::Slice(&a._data.internal[0])
-           .binaryEquals(arangodb::velocypack::Slice(&b._data.internal[0]));
+  switch (type) {
+    case arangodb::aql::AqlValue::VPACK_INLINE:
+      return arangodb::velocypack::Slice(a._data.inlineSliceMeta.slice)
+           .binaryEquals(arangodb::velocypack::Slice(b._data.inlineSliceMeta.slice));
+    case arangodb::aql::AqlValue::VPACK_INLINE_INT48:
+      // equal is equal. sign does not matter. So compare unsigned
+      return a._data.shortNumberMeta.data.int48.val == b._data.shortNumberMeta.data.int48.val;
+    case arangodb::aql::AqlValue::VPACK_INLINE_INT64:
+    case arangodb::aql::AqlValue::VPACK_INLINE_UINT64:
+    case arangodb::aql::AqlValue::VPACK_INLINE_DOUBLE:
+      // equal is equal. sign/endianess does not matter
+      return a._data.longNumberMeta.data.intLittleEndian.val == b._data.longNumberMeta.data.intLittleEndian.val;
```

Comment:
For double this is not correct! For example, `0.0 == -0.0` but the binary representation is different.

Reply:
I tend to agree, good point. But before here was used binaryEquals - this internally called memcmp on VPack. So having 2 slices one with double(0.0) and another with double(-0.0) we will detect non-equality.  So should we fix this or leave as is and keep old (erroneous) behaviour ? @jsteemann  WDYT ?

Reply:
Oh, ok - so it was broken before too! I wasn't of that that... :D

Reply:
It's true comparing by `==` or by using `.binaryEquals(...)` the values wouldn't compare equal.

I don't think this is a problem _here_ because the hash function for AqlValue previously already called `VPackSlice(...).hash()` (in contrast to `.normalizedHash()`).
And the equal_to function for AqlValue previously called `.binaryEquals()`, so it could also compare two logically equivalent values and conisider them non-equal.

For AqlValue, this is not a problem right now. AqlValues were previously compared only inside AqlItemBlock, and were placed as keys into an std::unordered_map to find duplicates. This was only important for AqlValues that used dynamic memory allocations anyway, but does not affect the "simple" AqlValues.

Because the unordered_map structure was changed a while ago and now uses the dynamic memory addresses as keys, it is possible that we don't even need the hash and equal_to functions for AqlValue anymore.
I can't tell yet if this is true, but I suggest we try removing the 2 functions and check if everything still compiles. If it does, we don't need to make any decision on the behavior.
And if this leads to compile errors, we should check from where it is called to make the decision.

Reply:
I think these functions are still necessary. But for what it's worth, the current behavior (already before this PR) is incorrect. This query:
```
FOR i in [0.0, -0.0, 1.0, -1.0]
  COLLECT x = i WITH COUNT INTO cnt
  RETURN [x, cnt]
```
produces this result:
```
[
  [-1,1],
  [0,1],
  [0,1],
  [1,1]
]
```